spylon_kernel/scala_magic.py

```diff
@@ -115,72 +166,61 @@ def eval(self, code, raw):
             else:
                 if res:
                     return TextOutput(res)
-        except ScalaException as e:
+        except ScalaException as ex:
+            # Get the kernel response so far
             resp = self.kernel.kernel_resp
+            # Wrap the exception for MetaKernel use
             resp['status'] = 'error'
-            tb = e.scala_message.split('\n')
+            tb = ex.scala_message.split('\n')
             first = tb[0]
             assert isinstance(first, str)
             eclass, _, emessage = first.partition(':')
-            # Include the entire traceback for notebook use
             return ExceptionWrapper(eclass, emessage, tb)
 
-    @option(
-        "-e", "--eval_output", action="store_true", default=False,
-        help="Use the retval value from the Scala cell as code in the kernel language."
-    )
-    def cell_scala(self, eval_output=False):
-        """
-        %%scala - evaluate contents of cell as Scala code
-
-        This cell magic will evaluate the cell (either expression or statement) as Scala code.
-
-        This will instantiate a scala interpreter prior to running the code.
-
-        The -e or --eval_output flag signals that the retval value expression will be used as code for the cell to be
-        evaluated by the host language.
-
-        Examples:
-            %%scala
-            val x = 42
+    def post_process(self, retval):
+        """Processes the output of one or stacked magics.
 
-            %%scala
-            import collections.mutable._
-            val y = mutable.Map.empty[Int, String]
+        Part of the MetaKernel API.
 
-            %%scala -e
-            retval = "'(this is code in the kernel language)"
+        Parameters
+        ----------
+        retval : any or None
+            Value from another magic stacked with this one in a cell
 
-            %%python -e
-            "'(this is code in the kernel language)"
+        Returns
+        -------
+        any
+            The received value if it's not None, otherwise the stored
+            `retval` of the last Scala code execution
         """
-        if self.code.strip():
-            if eval_output:
-                self.eval(self.code, False)
-                # self.code = str(self.env["retval"]) if ("retval" in self.env and
-                #                                         self.env["retval"] != None) else ""
-                self.retval = None
-                #self.env["retval"] = None
-                self.evaluate = True
-            else:
-                self.retval = self.eval(self.code, False)
-                #self.env["retval"] = None
-                self.evaluate = False
-
-    def post_process(self, retval):
         if retval is not None:
```

Comment:
I think this if/else block could be replaced with `return retval or self.retval`. Or at least drop the `else`

```python
if retval is not None:
    return retval
return self.retval
```